tests/Control/UserDefinedFormControllerTest.php

```diff
@@ -367,4 +372,56 @@ public function testRecipientSubjectMergeFields()
         // check emails
         $this->assertEmailSent('test@example.com', 'no-reply@example.com', 'Email Subject: Basic Value');
     }
+
+    /**
+     * Note: will temporarily create an image in the websites public/assets/.protected folder
+     */
+    public function testImageThumbnailCreated()
+    {
+        // ensure framework is installed and a functioning website exists so there's somewhere
+        // for the uploaded file to go
+        if (!class_exists(Upload_Validator::class) || !file_exists(ASSETS_PATH)) {
```

Comment:
We've got a dedicated TestAssetStore to handle this kind of situation.

[That's how you use it](https://github.com/silverstripe/silverstripe-assets/blob/68323e3572f828e5d06a77a49cce8f252c545306/tests/php/Storage/AssetStoreTest.php#L27-L33).

Reply:
Cheers, updated to use `TestAssetStore`